### PR TITLE
chore(RHTAPWATCH-765): pipeline-service can manage SpaceRequests

### PR DESCRIPTION
**Pull request by Omeramsc**
In order to create ephemeral namespaces for
testing konflux components, the pipeline-service
service account needs to be able to manage (create, delete, get) the SpaceRequest custom resource.

~Related PR:  https://github.com/codeready-toolchain/member-operator/pull/529~
Blocking: https://issues.redhat.com/browse/RHTAPWATCH-764

### CODE REVIEW

File: components/tekton-ci/base/appstudio-pipelines-runner-rolebinding.yaml [hunk 1]
```diff
@@ -10,3 +10,16 @@ subjects:
 - kind: ServiceAccount
   name: appstudio-pipeline
   namespace: tekton-ci
+---
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (gbenhaim) [inline components/tekton-ci/base/appstudio-pipelines-runner-rolebinding.yaml:14]: why this change is needed?

File: components/promotion/base/appstudio-pipelines-runner-rolebinding.yaml [hunk 2]
```diff
@@ -10,3 +10,16 @@ subjects:
 - kind: ServiceAccount
   name: appstudio-pipeline
   namespace: rhtap-promotion-staging
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
```
Comment (gbenhaim) [inline components/promotion/base/appstudio-pipelines-runner-rolebinding.yaml:15]: why this change is needed?

File: components/build-templates/base/e2e/rolebinding.yaml [hunk 3]
```diff
@@ -53,6 +53,19 @@ roleRef:
 ---
 kind: RoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: appstudio-pipelines-namespace-creator-rolebinding
+subjects:
```
Comment (gbenhaim) [inline components/build-templates/base/e2e/rolebinding.yaml:58]: why this change is needed?

Comment (Omeramsc) [reply components/build-templates/base/e2e/rolebinding.yaml:58]: I applied the namespace-creator to wherever there are configurations related to appstudio-pipelines for uniformity. should those three be removed? @gbenhaim

Comment (gbenhaim) [reply components/build-templates/base/e2e/rolebinding.yaml:58]: I suggest to wait with this PR for now. I don't think it's needed but maybe I'm missing something.

Comment (Omeramsc) [reply components/build-templates/base/e2e/rolebinding.yaml:58]: Speaking with the pipeline-services team, it seems like `appstudio-pipelines-runner-rolebinding.yaml` is linking the role (I added in https://github.com/codeready-toolchain/member-operator/pull/529) to the user itself, so I'm not sure if it can work without. I'll wait on the pipelines\build review anyway

**Comment by Omeramsc** [comment]
@gbenhaim I added the `appstudio-namespace-creator_role.yaml` to this PR as it wasn't relevant in the member-operator repository. 
We should probably not wait with this PR any longer as it is currently blocking https://issues.redhat.com/browse/RHTAPWATCH-764 as we don't have the means to test the Task without access to SpaceRequest.

File: components/tekton-ci/base/appstudio-pipelines-runner-rolebinding.yaml [hunk 4]
```diff
@@ -10,3 +10,16 @@ subjects:
...
+kind: RoleBinding
+metadata:
+  name: appstudio-pipelines-namespace-creator-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: appstudio-pipelines-namespace-creator
+subjects:
+- kind: ServiceAccount
+  name: appstudio-pipeline
```
Comment (mmorhun) [inline components/tekton-ci/base/appstudio-pipelines-runner-rolebinding.yaml:24]: I don't think that build pipeline should be able to create namespaces.

Comment (Omeramsc) [reply components/tekton-ci/base/appstudio-pipelines-runner-rolebinding.yaml:24]: I wonder though, even if removed here - if all `appstodio-pipeline` SA are gonna get created with the ability to create namespaces, wouldn't it give it this permission anyway? or am I missing something

**Comment by Omeramsc** [comment]
Not relevant for now, closing.
